modules/extensions-spring-support/src/main/java/org/mule/runtime/module/extension/internal/config/dsl/ComponentMessageProcessorObjectFactory.java

```diff
@@ -72,6 +72,7 @@ public P doGetObject() {
           .findFirst()
           .ifPresent(chain -> parameters.put(chain.getName(),
                                              new ProcessorChainValueResolver(nestedChain)));
+      nestedChain.setAnnotations(this.getAnnotations());
```

Comment:
is this ok? you are setting the annotations of the operation to one of the parameters of the operation.

Reply:
Without this, we have no way to get the location of the chain --> which we need to be the exact same when we create the new context. If not, we get a pretty big error regarding the sdk internal parameters (NPE, because they have a map that uses the location)

Reply:
put this explanation in a comment in the code